irispy/new_sji.py

```diff
@@ -0,0 +1,203 @@
+        such an interface is NDUncertainty - but isn’t mandatory. If the
+        uncertainty has no such attribute the uncertainty is stored as
+        UnknownUncertainty.
+        Defaults to None.
+
+    mask : any type, optional
+        Mask for the dataset. Masks should follow the numpy convention
+        that valid data points are marked by False and invalid ones with True.
+        Defaults to None.
+
+    extra_coords : iterable of `tuple`s, each with three entries
+        (`str`, `int`, `astropy.units.quantity` or array-like)
+        Gives the name, axis of data, and values of coordinates of a data axis
+        not included in the WCS object.
+
+    copy : `bool`, optional
+        Indicates whether to save the arguments as copy. True copies every
+        attribute before saving it while False tries to save every parameter
+        as reference. Note however that it is not always possible to save the
+        input as reference.
+        Default is False.
+
+    Examples
+    --------
+    >>> from irispy import sji
+    >>> from irispy.data import sample
+    >>> sji = read_iris_sji_level2_fits(sample.SJI_CUBE_1400)
+    """
+
+    def __init__(self, data, wcs, uncertainty=None, unit=None, meta=None,
+                 mask=None, extra_coords=None, copy=False, missing_axis=None,
+                 scaled=None):
+        """
+        Initialization of Slit Jaw Imager
+        """
+        # Set whether SJI data is scaled or not.
+        self.scaled = scaled
+        # Initialize SJI_NDCube.
+        super().__init__(data, wcs, uncertainty=uncertainty, mask=mask,
+                         meta=meta, unit=unit, extra_coords=extra_coords,
+                         copy=copy, missing_axis=missing_axis)
+
+    def __repr__(self):
+        return (
+            """
+    SJICube
+    ---------
+    Observatory:\t\t {obs}
+    Instrument:\t\t\t {instrument}
+    Bandpass:\t\t\t {bandpass}
+    Obs. Start:\t\t\t {date_start:{tmf}}
+    Obs. End:\t\t\t {date_end:{tmf}}
+    Instance Start:\t\t {instance_start:{tmf}}
+    Instance End:\t\t {instance_end:{tmf}}
+    Total Frames in Obs.:\t {frame_num}
+    IRIS Obs. id:\t\t {obs_id}
+    IRIS Obs. Description:\t {obs_desc}
+    Cube dimensions:\t\t {dimensions}
+    Axis Types:\t\t\t {axis_types}
+    """.format(obs=self.meta["TELESCOP"],
+               instrument=self.meta["INSTRUME"],
+               bandpass=self.meta["TWAVE1"],
+               date_start=self.meta["DATE_OBS"],
+               date_end=self.meta["DATE_END"],
+               instance_start=self.extra_coords["TIME"]["value"][0],
+               instance_end=self.extra_coords["TIME"]["value"][-1],
+               frame_num=self.meta["NBFRAMES"],
+               obs_id=self.meta["OBSID"],
+               obs_desc=self.meta["OBS_DESC"],
+               axis_types=self.world_axis_physical_types,
+               dimensions=self.dimensions, tmf=TIME_FORMAT))
```

Comment:
Here is my problem.

I tried to modify one date to `None` to check how the code works. By changing the date to `None` is the same result of a missing value for `DATE_OBS` or `DATE_END` on the fits file. So, on the meta, I have `'DATE_OBS' = None` for example. If I ask SJI class to use `__repr__`, the time will be transferred to the `TIME_FORMAT = config.get("general", "time_format")`.

The problem is that `None` can't be transferred as a time format. The code will crash if one date is missing in the FITS file.

Hoping to be clearer !

Reply:
Ah, yes I see.  I suggest in this case that you calculate the time values at the start of the start of the ```__repr__``` method, i.e. between current lines 94 and 95, and then put the result of that into the string.  This way you can use an if statement to handle cases where time is None.  So something like:
```
def __repr__(self):
    my_date = self.meta.get("MY_DATE", None)
    if my_date:
        my_string_date = # Convert to time string format here
    else:
        my_string_date = None
    return (
```
This way ```my_string_date``` is already converted to a time string (or is ```None```) and does not require formatting when generating the ```__repr__``` string itself.

Reply:
Is there a good function to do that ? I found some solutions but they are ugly ...

Reply:
To convert the time as a good string format used by NDCube.

Reply:
Solved